internal/compiler/passes/check_public_api.rs

```diff
@@ -7,26 +7,41 @@ use std::rc::Rc;
 
 use crate::diagnostics::{BuildDiagnostics, DiagnosticLevel};
 use crate::object_tree::{Component, Document, PropertyVisibility};
+use crate::CompilerConfiguration;
 
-pub fn check_public_api(doc: &Document, diag: &mut BuildDiagnostics) {
-    check_public_api_component(&doc.root_component, diag);
+pub fn check_public_api(
+    doc: &Document,
+    diag: &mut BuildDiagnostics,
+    compiler_config: &CompilerConfiguration,
+) {
+    check_public_api_component(&doc.root_component, diag, compiler_config);
     for (export_name, e) in &*doc.exports {
         if let Some(c) = e.as_ref().left() {
             if c.is_global() {
                 // This global will become part of the public API.
                 c.exported_global_names.borrow_mut().push(export_name.clone());
-                check_public_api_component(c, diag)
+                check_public_api_component(c, diag, compiler_config)
             }
         }
     }
 }
 
-fn check_public_api_component(root_component: &Rc<Component>, diag: &mut BuildDiagnostics) {
+fn check_public_api_component(
+    root_component: &Rc<Component>,
+    diag: &mut BuildDiagnostics,
+    compiler_config: &CompilerConfiguration,
+) {
     let mut root_elem = root_component.root_element.borrow_mut();
     let root_elem = &mut *root_elem;
     let mut pa = root_elem.property_analysis.borrow_mut();
     root_elem.property_declarations.iter_mut().for_each(|(n, d)| {
-        if d.property_type.ok_for_public_api() {
+        if d.property_type == crate::langtype::Type::ComponentFactory && !compiler_config.support_component_containers {
+            diag.push_diagnostic(
+                 format!("Properties of type {} are not supported in this compiler configuration. The property will not be exposed", d.property_type),
```

Comment:
We should not do the check here for public API, but for anything.  the component-factory type shouldn't be allowed at  all and shouldn't be completed by the LSP.
So i'd go as far as removing it from the type register if it's not enabled.

(Even if we get an eror "don't exist" instead of "not allowed in the configuration")